### PR TITLE
added material table for AdminModule. Transfer new-news.component to …

### PR DESCRIPTION
**Pull request by ShpakKate**
…UserModule. Fixes errors. Creation server for news

### CODE REVIEW

**Comment by burymm** [comment]
@ShpakKate please resolve conflicts before

**Comment by Dewlar** [comment]
all merge problems have been resolved in this branch.
@ShpakKate feels good ;)

File: src/app/admin/admin-layout/admin-layout.component.ts [hunk 1]
```diff
@@ -12,9 +12,8 @@ export class AdminLayoutComponent {
 
   constructor(private router: Router, private auth: AuthenticationService) {}
 
-  logout(event: any) {
+  logout(event: MouseEvent) {
```
Comment (Dewlar) [inline src/app/admin/admin-layout/admin-layout.component.ts:15]: Unused parameter data

File: src/app/components/newsitem/newsitem.component.ts [hunk 2]
```diff
@@ -14,28 +14,43 @@ export class NewsitemComponent implements OnInit {
...
-    return this.item?.full?.slice(0, 40) !== this.item.full;
+  hidingBtn() {
+    return this.a !== this.item.full;
   }
 
   deleteNews(item: News) {
-    this.dataService.deleteNews(item);
+    this.newsList$
+      .pipe(
+        switchMap(users => this.dataService.deleteNews(item)),
```
Comment (Dewlar) [inline src/app/components/newsitem/newsitem.component.ts:50]: users is unused parameter data, so must be deleted

File: src/app/admin/components/list-of-users/list-of-users.component.ts [hunk 3]
```diff
@@ -1,48 +1,39 @@
...
+  dataToDisplay = [this.userList$];
   notAdmin = false;
 
-  constructor(private auth: AuthenticationService) {}
+  @ViewChild(MatPaginator) paginator!: MatPaginator;
+  @ViewChild(MatSort) sort!: MatSort;
+
+  constructor(private auth: AuthenticationService, public dialog: MatDialog) {
+    this.auth.getList().pipe(
+      tap(data => {
```
Comment (Dewlar) [inline src/app/admin/components/list-of-users/list-of-users.component.ts:29]: `data` is unused parameter data
but all of the `tap`  operator is unnecessary so can be removed

File: src/app/admin/components/list-of-users/list-of-users.component.ts [hunk 4]
```diff
@@ -1,48 +1,39 @@
...
+      })
+    );
+  }
 
   ngOnInit() {
     this.loadData().subscribe();
-    this.getElementData();
   }
 
   loadData() {
```
Comment (Dewlar) [inline src/app/admin/components/list-of-users/list-of-users.component.ts:39]: method for nothing

**Review by Dewlar — CHANGES_REQUESTED** [review]
need to make a few minor edits
otherwise good

File: src/app/admin/components/list-of-users/list-of-users.component.ts (hunk 4 above)
Comment (Dewlar) [reply src/app/admin/components/list-of-users/list-of-users.component.ts:39]: sorry, wrong marker
i mean `loadData(){}` - it is the method for nothing
it is empty

File: src/app/admin/components/list-of-users/list-of-users.component.ts [hunk 5]
```diff
@@ -1,48 +1,39 @@
...
   notAdmin = false;
 
-  constructor(private auth: AuthenticationService) {}
+  @ViewChild(MatPaginator) paginator!: MatPaginator;
+  @ViewChild(MatSort) sort!: MatSort;
+
+  constructor(private auth: AuthenticationService, public dialog: MatDialog) {
+    this.auth.getList().pipe(
+      tap(data => {
+        console.log(this.dataSource.data);
```
Comment (burymm) [inline src/app/admin/components/list-of-users/list-of-users.component.ts:30]: also console log can be removed from prod code

**Pull request closed without merging**